blogposts/2021/optimizing-a-code-intelligence-commit-graph-1.md

```diff
@@ -0,0 +1,257 @@
+---
+title: "Optimizing a code intelligence commit graph (Part 1)"
+author: Eric Fritz
+authorUrl: https://eric-fritz.com
+description: "Read about our journey to enabling Sourcegraph to respond to requests on commits missing an index quickly, with precise results."
+publishDate: 2021-05-24T18:00-07:00
+tags: [blog]
+slug: optimizing-a-code-intel-commit-graph
+heroImage: /blog/optimizing-code-intelligence-commit-graph.png
+socialImage: /blog/optimizing-code-intelligence-commit-graph.png
+published: true
+---
+
+![Optimizing code intelligence commit graph graphic](/blog/optimizing-code-intelligence-commit-graph.png)
+
+Sourcegraph's [Code Intelligence](https://about.sourcegraph.com/handbook/engineering/code-intelligence) team builds tools and services that provide contextual information around code. These enable users to perform fast, comprehensive, and accurate code navigation, to surface dependency relationships across projects, repositories, and languages, and to index and keep code up to date across all languages. In this post I'll dive into how Sourcegraph can resolve code intelligence queries using data from older commits when data on the requested commit is not yet available. In Part 2 I'll cover how we tackled scalability of this feature. 
```

Comment:
@efritz I added this to contextualize for people who might read the post who haven't yet heard of SG. Feel free to shorten/tweak

Reply:
```suggestion
Sourcegraph's [Code Intelligence](https://about.sourcegraph.com/handbook/engineering/code-intelligence) team builds tools and services that provide contextual information around code. These enable users to perform fast, comprehensive, and accurate code navigation, and to surface dependency relationships across projects, repositories, and languages. In this post I'll dive into how Sourcegraph can resolve code intelligence queries using data from older commits when data on the requested commit is not yet available. In Part 2, we'll cover how additional scalability concerns presented themselves and how we tackled them.
```